BoxContentSDK/BoxContentSDK/Requests/BOXFolderCreateRequest.m

```diff
@@ -38,50 +60,120 @@ - (BOXAPIOperation *)createOperation
     NSDictionary *bodyDictionary = @{BOXAPIObjectKeyParent : @{BOXAPIObjectKeyID : self.parentFolderID},
                                      BOXAPIObjectKeyName : self.folderName};
 
-    BOXAPIJSONOperation *JSONOperation = [self JSONOperationWithURL:URL
-                                                         HTTPMethod:BOXAPIHTTPMethodPOST
-                                              queryStringParameters:queryParameters
-                                                     bodyDictionary:bodyDictionary
-                                                   JSONSuccessBlock:nil
-                                                       failureBlock:nil];
-    [self addSharedLinkHeaderToRequest:JSONOperation.APIRequest];
-
-    return JSONOperation;
+    if (self.associateId != nil) {
+        BOXAPIDataOperation *dataOperation = [self dataOperationWithURL:URL
+                                                             HTTPMethod:BOXAPIHTTPMethodPOST
+                                                  queryStringParameters:queryParameters
+                                                         bodyDictionary:bodyDictionary
+                                                           successBlock:nil
+                                                           failureBlock:nil
+                                                            associateId:self.associateId];
+        
+        NSString *requestDirectory = [BOXContentClient defaultClient].tempCacheDir;
+        if (requestDirectory == nil) {
+            requestDirectory = NSTemporaryDirectory();
+        }
+        
+        NSString *destinationPath = [requestDirectory stringByAppendingPathComponent:self.associateId];
```

Comment:
Shouldn't there be an intermediate folder between tmp (or the baseline one for content client) and the item ID?